### PR TITLE
Catch a bug in mimetype library

### PR DESCRIPTION
**Pull request by kyleknap**
Issue was that on windows some MIME types in the Windows library could not be encoded using the system default encoding. This has been fixed in later versions of python, but we should not hard fail if the user is using a python version without the fix. Instead just catch the exception and not set the content type.

Fixes https://github.com/aws/aws-cli/issues/774

cc @jamesls @mtdowling @rayluo @JordonPhillips

### CODE REVIEW

File: tests/functional/s3/test_cp_command.py [hunk 1]
```diff
@@ -163,6 +163,15 @@ def test_no_metadata_directive_for_non_copy(self):
...
-if __name__ == "__main__":
-    unittest.main()
+    def test_cp_succeeds_with_mimetype_errors(self):
+        full_path = self.files.create_file('foo.txt', 'mycontent')
+        cmdline = '%s %s s3://bucket/key.txt' % (self.prefix, full_path)
+        self.parsed_responses = [
+            {'ETag': '"c8afdb36c52cf4727836669019e69222"'}]
+        with mock.patch('mimetypes.guess_type') as mock_guess_type:
+            # This should throw a UnicodeDecodeError.
+            mock_guess_type.side_effect = lambda x: b'\xe2'.decode('ascii')
```
Comment (kyleknap) [inline tests/functional/s3/test_cp_command.py]: I would be up for suggestions on a better way to throw the ``UnicodeDecodeError``. I was having troubles instantiate directly. Think its due to the fact that is a built-in C class.

File: awscli/customizations/s3/fileinfo.py [hunk 2]
```diff
@@ -266,7 +270,22 @@ def _handle_upload(self, body):
...
+        # https://bugs.python.org/issue9291
+        #
+        # So instead of hard failing, just log the issue and fall back to the
+        # default guessed content type of None.
+        except UnicodeDecodeError:
+            LOGGER.debug(
+                'Unable to guess content type for %s due to '
+                'UnicodeDecodeError: ', filename, exc_info=True
+            )
         if guessed_type is not None:
```
Comment (mtdowling) [inline awscli/customizations/s3/fileinfo.py]: I think this should be moved in the try block before the except block. We only need to check if `guessed_type` is not None when a unicode exception was not raised.

Comment (kyleknap) [inline awscli/customizations/s3/fileinfo.py]: That would work to. I default it to ``None`` before the try except. I would be fine with either.

**Comment by mtdowling** [comment]
Looks good. One small comment.

**Comment by rayluo** [comment]
:ship:

**Comment by mtdowling** [comment]
:ship:

**Pull request merged**